app/models/runtime/buildpack.rb

```diff
@@ -1,23 +1,35 @@
+require 'cloud_controller/diego/lifecycles/lifecycles'
+
 module VCAP::CloudController
   class Buildpack < Sequel::Model
-    plugin :list
+    plugin :list, scope: :lifecycle
+    plugin :after_initialize
 
-    export_attributes :name, :stack, :position, :enabled, :locked, :filename
-    import_attributes :name, :stack, :position, :enabled, :locked, :filename, :key
+    export_attributes :name, :stack, :position, :enabled, :locked, :filename, :lifecycle
+    import_attributes :name, :stack, :position, :enabled, :locked, :filename, :lifecycle, :key
 
     PACKAGE_STATES = [
       CREATED_STATE = 'AWAITING_UPLOAD'.freeze,
       READY_STATE = 'READY'.freeze
     ].map(&:freeze).freeze
 
+    def after_initialize
+      self.lifecycle ||= Config.config.get(:default_app_lifecycle)
+    end
+
     one_to_many :labels, class: 'VCAP::CloudController::BuildpackLabelModel', key: :resource_guid, primary_key: :guid
     one_to_many :annotations, class: 'VCAP::CloudController::BuildpackAnnotationModel', key: :resource_guid, primary_key: :guid
 
     add_association_dependencies labels: :destroy
     add_association_dependencies annotations: :destroy
 
-    def self.list_admin_buildpacks(stack_name=nil)
+    def self.user_visibility_filter(_user)
+      full_dataset_filter
+    end
+
+    def self.list_admin_buildpacks(stack_name=nil, lifecycle=Config.config.get(:default_app_lifecycle))
       scoped = exclude(key: nil).exclude(key: '')
+      scoped = scoped.filter(lifecycle:)
```

Comment:
I think we need to remove this line (or the default arg).  This hides all buildpacks of the non-default type on the list endpoint

Reply:
I guess this is intentional: https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0031-system-cnb.md#proposal

It's very confusing though, and breaks the CLI if you are trying to update a buildpack that is not of the default lifecycle

Reply:
@Gerg pointed out to me that line in the RFC is about POST, not list.  @pbusko was it your intention to have a default filter on the list endpoint?

Reply:
this filter is for backward compatibility, since the list endpoint has additional `lifecycle` query parameter (just like other buildpack related endpoints).
If we remove it, current versions of CF CLI will receive a soup of buildpacks with mixed lifecycles. I believe there's no way around it, it has to be filtered